packages/squiggle-lang/src/rescript/Reducer/Reducer_CodeTree/Reducer_CodeTree.res

```diff
@@ -0,0 +1,91 @@
+module BuiltIn = Reducer_Dispatch_BuiltIn
+module T = Reducer_CodeTree_T
+module CTV = Reducer_Extension.CodeTreeValue
+module MJ = Reducer_MathJs_Parse
+module MJT = Reducer_MathJs_ToCodeTree
```

Comment:
It is because of the data flow.

MathJs parser  depends on CodeTree to convert its parse to codeTree. CodeTree should not know about MathJs parser.

But codeTreeValue depends on MathJs because we are importing the math (log, cos, sin, >, >=, <, etc)

So parsing and evaluating has a different direction of dependency

Reply:
Would it be possible to move the MathJS execution into the Extension; for example, the ``Reducer_Extension_ReducerLibrary``

Reply:
ReducerExtention has become ReducerInterface.   
ReducerInterface is what you need to integrate distributions. So MathJs has no place there.

Your uneasiness problem will be gone when we trash MathJs parser.   There are actually 2 different things. 
- MathJs parser => MathJs
- MathJs maths library => MathJs

I am using those 2 things as separate things. Not as one and only one mighty MathJs. So there is MathJs and MathJs. There are 2 dependencies for totally 2 different purposes. However, MathJs and MathJs is only MathJs. So this is creating a confusion. I am separating that in the code.

One day, the parser will be gone and we will happily live ever after with the MathJs maths library.

Reply:
RESOLVED with a Todo. I will provide you a diagram